### PR TITLE
Add new revamped dropdown widget

### LINKED ISSUE #8228: Dropdown component 2.0
PRD
[Dropdown -New](https://app.clickup.com/37484951/v/dc/13qycq-2704/13qycq-14576)
[Dropdown (Deprecated)](https://app.clickup.com/37484951/v/dc/13qycq-2704/13qycq-22656)
Design [here](https://www.figma.com/file/mnyECAMd6Wc7qRIDyYmYfI/Components-redesign-01?type=design&node-id=8160-139783&mode=design&t=gUdgnTdSTeyvC9J0-4)

### PR DESCRIPTION
**Pull request by nakulnagargade**
Closes #8228.

### CODE REVIEW

**Comment by johnsoncherian** [comment]
Search box loses focus as soon as user types. Prevent user from typing more than one word

https://github.com/ToolJet/ToolJet/assets/57667706/3026266a-3899-4ab9-ac6c-da1e7d6dfd08

File: frontend/src/Editor/Components/DropdownV2/DropdownV2.jsx [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+  setExposedVariable,
+  fireEvent,
+  darkMode,
+  onComponentClick,
+  id,
+  component,
+  exposedVariables,
+  dataCy,
+}) => {
+  let {
```
Comment (johnsoncherian) [inline frontend/src/Editor/Components/DropdownV2/DropdownV2.jsx:161]:
```suggestion
  const {
```

File: frontend/src/Editor/WidgetManager.jsx [hunk 2]
```diff
@@ -101,12 +103,14 @@ export const WidgetManager = function WidgetManager({ componentTypes, zoomLevel,
     ];
     const integrationItems = ['Map'];
     const layoutItems = ['Container', 'Listview', 'Tabs', 'Modal'];
+    const legacyItems = ['ToggleSwitchLegacy', 'DropdownLegacy'];
```
Comment (johnsoncherian) [inline frontend/src/Editor/WidgetManager.jsx:106]: Can we move this to a config file inside Widget Manager?
Move all of constants if possible

File: frontend/src/Editor/Components/DropdownV2/DropdownV2.jsx [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+import TriangleUpCenter from '@/_ui/Icon/solidIcons/TriangleUpCenter';
+import cx from 'classnames';
+import { useEditorStore } from '@/_stores/editorStore';
+import Loader from '@/ToolJetUI/Loader/Loader';
+import { has, isObject, pick } from 'lodash';
+const tinycolor = require('tinycolor2');
+import SolidIcon from '@/_ui/Icon/SolidIcons';
+import './dropdownV2.scss';
+
+const { ValueContainer, SingleValue, Placeholder, DropdownIndicator, ClearIndicator, MenuList, Menu } = components;
```
[the remark this answers is not in the document]
Comment (nakulnagargade) [reply frontend/src/Editor/Components/DropdownV2/DropdownV2.jsx]: Done

File: frontend/src/Editor/WidgetManager.jsx (hunk 2 above)
Comment (nakulnagargade) [reply frontend/src/Editor/WidgetManager.jsx]: Done

**Comment by nakulnagargade** [comment]
> Search box loses focus as soon as user types. Prevent user from typing more than one word
> 
>  Screen.Recording.2024-05-23.at.4.37.47.PM.mov

This is fixed.

**Comment by nakulnagargade** [comment]
Closing this PR since this changes are also available in https://github.com/ToolJet/ToolJet/pull/9837.

**Pull request closed without merging**